py/desitarget/mock/mockmaker.py

```diff
@@ -3411,6 +3425,23 @@ def make_spectra(self, data=None, indx=None, seed=None, no_spectra=False):
                     qso_flux[these, :] = qso_flux1
 
             meta['SUBTYPE'][:] = 'LYA'
+            ##Added DLAs before lya forest trasnmission.
+            if add_dlas:
+                min_lya_z = np.min(skewer_wave/lambda_RF_LYA - 1)
+                for ii in range(len(skewer_meta['MOCKID'])):
+                    # quasars with z < min_z will not have any DLA in spectrum
+                    if min_lya_z>skewer_meta['Z'][ii]: continue
+                    # skewer ID
+                    idd=skewer_meta['MOCKID'][ii]
+                    dlas=[]
+                    for dla in dla_info[dla_info['MOCKID']==idd]:
+                        # Adding only DLAs with z < zqso
+                        if dla['Z_DLA_RSD']>=skewer_meta['Z'][ii]: continue
+                        dlas.append(dict(z=dla['Z_DLA_RSD'],N=dla['N_HI_DLA'],dlaid=dla['DLAID']))
+                    transmission_dla = dla_spec(skewer_wave,dlas)
+                    if len(dlas)>0:
+                        skewer_trans[ii] = transmission_dla * skewer_trans[ii]
+                        objmeta['DLA_MOCKID'][ii]=idd
```

Comment:
@andreufont notice here I'm saving the  MOCKID not the DLAID, because there could be more than one DLA in a given skewer and it was not easy to save multiple ids in the same object. But I've tested and with this is easy to retrieve the truth DLA catalog... but is as good as setting a True/False variable... What you think would be preferable?

Reply:
I would use someting that is easy/clear to code/document, and that is good enough for what those working in DLA finders need. 

I'd imagine that what they need is a way to answer the following question: for a particular simulated spectra, does it contain DLAs? And if so, what are the redshifts and column densities for each DLA?

Ideally all the information would be in the truth file, but we could have a different truth_dla file or similar if needed, I guess?